packages/oauth2/src/server/routes.ts

```diff
@@ -0,0 +1,220 @@
+import {Application, NextFunction, Request, Response} from 'express'
+import Provider from 'oidc-provider'
+import {urlencoded} from 'express'
+import isEmpty from 'lodash/isEmpty'
+import querystring from 'querystring'
+import {inspect} from 'util'
+
+import * as assert from 'assert'
+import {MongoDBAdapter} from '@wepublish/api-db-mongodb/lib'
+
+const body = urlencoded({extended: false})
+
+const keys = new Set()
+const debug = (obj: any) =>
+  querystring.stringify(
+    Object.entries(obj).reduce((acc: any, [key, value]) => {
+      keys.add(key)
+      if (isEmpty(value)) return acc
+      acc[key] = inspect(value, {depth: null})
+      return acc
+    }, {}),
+    '<br/>',
+    ': ',
+    {
+      encodeURIComponent(value) {
+        return keys.has(value) ? `<strong>${value}</strong>` : value
+      }
+    }
+  )
+
+export function routes(app: Application, provider: Provider, dbAdapter: MongoDBAdapter): void {
+  //const { constructor: { errors: { SessionNotFound } } } = provider;
+
+  app.use((req, res, next) => {
+    const orig = res.render
+    // you'll probably want to use a full blown render engine capable of layouts
```

Comment:
We should consider using some pluggable React components for this.

Reply:
Agree. Also created a new issue for that. https://wepublish.atlassian.net/browse/WEP-67